configuration/src/main/java/io/camunda/configuration/Cluster.java

```diff
@@ -169,29 +159,11 @@ public void setInitialContactPoints(final List<String> initialContactPoints) {
   }
 
   public Integer getNodeId() {
-    if (dynamicNodeId.getType() != Type.NONE && nodeId != null) {
-      throw new UnifiedConfigurationException(
-          String.format(
-              "Both %s.dynamic-node-id and %s.node-id are set at the same time. Only one of them can be set as they are mutually exclusive",
-              PREFIX, PREFIX));
-    }
-    if (dynamicNodeId.getType() == Type.NONE) {
-      final var id =
-          UnifiedConfigurationHelper.validateLegacyConfiguration(
-              PREFIX + ".node-id",
-              nodeId,
-              Integer.class,
-              UnifiedConfigurationHelper.BackwardsCompatibilityMode.SUPPORTED,
-              Set.of(legacyPropertiesMap.get("nodeId")));
-
-      return Optional.ofNullable(id).orElse(0);
-    } else {
-      return null;
-    }
+    return nodeIdProvider.staticConfig().getNodeId();
   }
 
   public void setNodeId(final int nodeId) {
-    this.nodeId = nodeId;
+    nodeIdProvider.staticConfig().setNodeId(nodeId);
```

Comment:
:question: What is the expected user configuration property here? `camunda.cluster.nodeid` or `camunda.cluster.node-id-provider.static.node-id`? If it is the second, do we want to deprecate the previous one?